JS/Lesson33/goksel/my-app/src/modules/todos/TodoProvider.jsx

```diff
@@ -0,0 +1,46 @@
+import { createContext, useContext, useReducer } from "react";
+
+export const todoData = [
+    { id: 1, title: "Do dishes", completed: false },
+    { id: 2, title: "Do homework", completed: false },
+    { id: 3, title: "Go running", completed: false },
+    { id: 4, title: "Do dishes 2", completed: false },
+    { id: 5, title: "Do homework 3", completed: false },
+    { id: 6, title: "Go running 4", completed: false },
+];
+
+
+export const TODO_ACTIONS = {
+    todoCompleted : "todo_completed"
+}
+
+export const TodoContext = createContext();
+export const TodoDispatchContext = createContext();
+
+export const TodoProvider = ({children, initialState}) => {
+
+    const [todo, dispatch] = useReducer(todoReducer, initialState ?? {} );
+
+    return (
+        <TodoContext.Provider value={todo}>
+            <TodoDispatchContext.Provider value={dispatch}>
+                {children}
+            </TodoDispatchContext.Provider>
+        </TodoContext.Provider>
+    )
+}
+
+function todoReducer(state, action) {
```

Comment:
When writing reducer functions, action is commonly used in the following format:
```
{
 type: 'action_type',
 payload: {}
}
```

Using an object for payload allows you to use different kinds of data in the reducer function. E.g. once you add functionality to add a new todo item, you'd need a todo title instead of todo id.